packages/node/src/parseForPolicy.js

```diff
@@ -114,54 +106,100 @@ async function parseForPolicy({
   })
 }
 
-function makeResolveHook({ projectRoot, resolutions = {}, canonicalNameMap }) {
+/**
+ *
+ * @param {string} rootDir
+ * @param {Record<string, string>} canonicalNameMap
+ * @param {Record<string, string>} [resolutions]
+ * @returns {ResolveHook}
+ */
+function makeResolutionResolveHook(rootDir, canonicalNameMap, resolutions) {
```

Comment:
I extracted this because I thought I would want two separate `makeResolveHook` implementations, but that turned out not to be true.  I left it as its own function just for the sake of splitting up a big function.